File: Java-Core/src/main/java/presto/type/DocumentType.java
package presto.type;

import presto.error.SyntaxError;
import presto.runtime.Context;
import presto.value.Document;

public class DocumentType extends NativeType {
	
	static DocumentType instance = new DocumentType();
	
	public static DocumentType instance() {
		return instance;
	}
	
	private DocumentType() {
		super("Document");
	}

	@Override
	public Class<?> toJavaClass() {
		return Document.class;
	}
	
	@Override
	public boolean isAssignableTo(Context context, IType other) {
		return (other instanceof DocumentType) || (other instanceof AnyType);
	}
	
	@Override
	public IType checkMember(Context context, String name) throws SyntaxError {
		return AnyType.instance();
	}
	

}

File: Java-CodeServer/src/main/resources/js/prompto/type/DocumentType.js
var NativeType = require("./NativeType").NativeType;
var AnyType = require("./AnyType").AnyType;

function DocumentType() {
	NativeType.call(this, "Document");
	return this;
}

DocumentType.prototype = Object.create(NativeType.prototype);
DocumentType.prototype.constructor = DocumentType;

DocumentType.instance = new DocumentType();


/*
@Override
public Class<?> toJavaClass() {
	return Document.class;
}

@Override
public boolean isAssignableTo(Context context, IType other) {
	return (other instanceof DocumentType) || (other instanceof AnyType);
}
*/

DocumentType.prototype.checkMember = function(context, name) {
	return AnyType.instance;
};

exports.DocumentType = DocumentType;
